File: java/po/world_simulator/entities/plants/Grass.java
package po.world_simulator.entities.plants;

import java.util.Map;

import po.world_simulator.Vector2;
import po.world_simulator.entities.Plant;

public class Grass extends Plant {
    public Grass(Vector2 position, int age) {
        super(0, position, "trawa:)", age);
    }

    public Grass(Vector2 position) {
        this(position, 0);
    }

    public Grass(Map<String, String> entityData){
        this(
            new Vector2(entityData.get("position")),
            Integer.parseInt(entityData.get("age"))
        );
    }
}

File: python/po/world_simulator/entities/plants/Grass.py
from po.world_simulator.entities.Plant import Plant
from po.world_simulator.Vector2 import Vector2

class Grass(Plant):
    def __init__(self, position_or_data, age=0):
        if isinstance(position_or_data, dict):
            super().__init__(0, Vector2(position_or_data.get("position")), "trawa:)", int(position_or_data.get("age", 0)))
        else:
            super().__init__(0, position_or_data, "trawa:)", age)